app/api/logging/decodelog.py

```diff
@@ -72,20 +74,21 @@ def format_line(
     level: str,
     message: str,
     extra: Mapping[str, str],
+    message_width: int = DEFAULT_MESSAGE_WIDTH,
 ) -> str:
     """Format log fields as a coloured string."""
-    return "%s  %s%-36s%s %-28s %s%-8s %-80s %s%s%s" % (
-        format_datetime(created),
-        colour_for_name(name),
-        name,
-        RESET,
-        func_name,
-        colour_for_level(level),
-        level,
-        message,
-        BLUE,
-        format_extra(extra),
-        RESET,
+    return "{created}  {logger_color}{logger_name:<36}{reset_color} {func_name:<28} {level_color}{level:<8} {message} {extra_color}{extra}{reset_color}".format(
+        created=format_datetime(created),
+        logger_color=colour_for_name(name),
+        logger_name=name,
+        reset_color=RESET,
```

Comment:
Not for this PR, just thinking. Since the way color is implemented is more or less `<START COLOR> txt <STOP COLOR>` would it be clearer here to instead have the color functions more or less do that?

Something like:
```
def colorized_text(text: str, color: str) -> str:
      return f"{color}{text}{RESET}"
```
The width would make that a bit trickier but probably workable.

Reply:
Yeah I had a similar thought! I think that would make the code a bit easier to read too.